implementations/micrometer-registry-opentelemetry/src/main/java/io/micrometer/opentelemetry/OpenTelemetryFunctionCounter.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright 2020 VMware, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.opentelemetry;
+
+import java.util.function.ToDoubleFunction;
+
+import io.micrometer.core.instrument.cumulative.CumulativeFunctionCounter;
+import io.opentelemetry.common.Labels;
+import io.opentelemetry.metrics.AsynchronousInstrument;
+import io.opentelemetry.metrics.DoubleSumObserver;
+
+public class OpenTelemetryFunctionCounter<T> extends CumulativeFunctionCounter<T> {
+    final DoubleSumObserver observer;
+
+    public OpenTelemetryFunctionCounter(Id id, T obj, ToDoubleFunction<T> value, DoubleSumObserver observer) {
+        super(id, obj, value);
+        this.observer = observer;
+        this.observer.setCallback(this::update);
+    }
+
+    public void update(AsynchronousInstrument.DoubleResult result) {
+        result.observe(count(), Labels.empty());
```

Comment:
does the value of `count()` get reset ever, or is it monotonically increasing?

Reply:
count as defined in the CumulativeFunctionCounter: 
```java
    @Override
    public double count() {
        T obj2 = ref.get();
        return obj2 != null ? (last = f.applyAsDouble(obj2)) : last;
    }
```
The FunctionCounter approach is not doing any aggregation itself: it is observing the value observed by applying the specified function on the specified object. To tie back to the docs: 

> Micrometer also provides a more infrequently used counter pattern that tracks a monotonically increasing function (a function that stays the same or increases over time, but never decreases). Some monitoring systems, like Prometheus, push cumulative values for counters to the backend, but others publish the rate at which a counter is incrementing over the push interval. By employing this pattern, you allow the Micrometer implementation for your monitoring system to choose whether to rate normalize the counter or not and your counter remains portable across different types of monitoring systems. 

So, it's up to this registry to decide how to work w/ the OTel SDK to do the right thing

Reply:
ok, then maybe an UpDownSumObserver might be better, if we can't guarantee monotonicity of the `count()` value.